src/EditorFeatures/Core.Wpf/Options/EditorConfigDocumentOptionsProvider.cs

```diff
@@ -119,8 +128,24 @@ private void Workspace_DocumentOpened(object sender, DocumentEventArgs e)
         private Task<ICodingConventionContext> GetConventionContextAsync(string path, CancellationToken cancellationToken)
         {
             return IOUtilities.PerformIOAsync(
-                () => _codingConventionsManager.GetConventionContextAsync(path, cancellationToken),
+                async () =>
+                {
+                    var context = await _codingConventionsManager.GetConventionContextAsync(path, cancellationToken).ConfigureAwait(false);
+                    context.CodingConventionsChangedAsync += OnCodingConventionsChangedAsync;
+
+                    return context;
+                },
                 defaultValue: EmptyCodingConventionContext.Instance);
         }
+
+        private Task OnCodingConventionsChangedAsync(object sender, CodingConventionsChangedEventArgs arg)
```

Comment:
❓ @olegtk Is this event ever invoked more often than one invocation for one change of .editorconfig on disk?

Reply:
if you are asking whether changing 1 editconfig file that affects multiple files will cause multiple events to be fired. then I tested it, and they do fire events for each file that got affected by the edit config file.

Reply:
there should be 1 event per file change (as raised by the VS file watcher service), and it's fired on each affected context instance.